### PR TITLE
[IMP] website: add `s_faq_horizontal` snippet

### PR DESCRIPTION
**Pull request by stefanorigano**
This PR adds a new snippet that provides a simple layout aiming to organise and showcase topics, like a FAQ list,  a walkthrough guide or product's features.
The Topics' titles are sticky by default to guide the visitor through the reading. 

| Preview |
|--------|
| image | 

task-4094403
part of task-4077427

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/website/views/snippets/s_faq_horizontal.xml
Comment (Brieuc-brd) [inline addons/website/views/snippets/s_faq_horizontal.xml:8]: Defining multiple rows will limit the use of the grid feature only on the first row, so only on your heading.
Was it on purpose ? If not, I would prevent the use of grid for this snippet 🙂 .

File: addons/website/views/snippets/s_faq_horizontal.xml
Comment (Brieuc-brd) [inline addons/website/views/snippets/s_faq_horizontal.xml:15]: I don't think `col-12` is necessary.
Same for the others below 🙂

File: addons/website/views/snippets/s_faq_horizontal.xml
Comment (Brieuc-brd) [inline addons/website/views/snippets/s_faq_horizontal.xml:18]: `overflow-auto` after `pb-lg-3` 😬
Same for the others below

File: addons/website/views/snippets/s_faq_horizontal.xml
Comment (Brieuc-brd) [inline addons/website/views/snippets/s_faq_horizontal.xml:24]: Unfortunately, if you want to take advantage of the WB to define the image width, you'll have to do in inline css 😬 
```suggestion
                            <img src="/web/image/website.s_faq_horizontal_default_image_1" class="img img-fluid rounded" alt="" style="width: 100% !important"/>
```
Capture d’écran 2024-08-12 à 18 07 29

File: addons/website/static/src/img/snippets_demo/s_faq_horizontal_1.jpg
Comment (Brieuc-brd) [inline addons/website/static/src/img/snippets_demo/s_faq_horizontal_1.jpg:1]: I'm not convinced by this image, too dark ?
What about this one ? https://unsplash.com/photos/brown-ceramic-teacup-XtUd5SiX464
Capture d’écran 2024-08-12 à 18 29 02

**Review by Brieuc-brd — CHANGES_REQUESTED** [review]
Looks nice ! Good job @stefanorigano 👍 
I've left a few comments below.

Also, don't forget to add a screenshot of this snippet in your PR description ;)

File: addons/website/views/snippets/s_faq_horizontal.xml
Comment (stefanorigano) [reply addons/website/views/snippets/s_faq_horizontal.xml:15]: Replaced with `col-lg-12` 👍 
A `row > col-` structure it's needed to trigger the clone/remove features + other useful options like padding, background, visibility etc...

File: addons/website/views/snippets/s_faq_horizontal.xml
Comment (Brieuc-brd) [inline addons/website/views/snippets/s_faq_horizontal.xml:43]: nitpicking :)
```suggestion
                            <div data-snippet="s_chart" data-name="Chart" class="s_chart o_draggable" data-type="line" data-legend-position="top" data-tooltip-display="true" data-stacked="false" data-border-width="1" data-data="{'labels':['v15','v16','v17','v18'],'datasets':[{'label':'Improvements','data':['12','24','48','200'],'backgroundColor':'o-color-1','borderColor':'o-color-1'}]}" data-max-value="250" data-ticks-min="NaN" data-ticks-max="250">
```

**Review by Brieuc-brd — CHANGES_REQUESTED** [review]
Thanks @stefanorigano 👍 
Just a little nitpicking below 😄 

I also found a very specific bug:
If you scroll enough to trigger the sticky position of the title, then change its background: this will remove the `top` calculation preventing the title from being hidden by the fixed header 🤔 :
https://www.awesomescreenshot.com/video/30697559?key=cb46ee9e65502a7225b3785d834e227b

Courage 💪

File: addons/website/views/snippets/s_faq_horizontal.xml
Comment (stefanorigano) [reply addons/website/views/snippets/s_faq_horizontal.xml:43]: Oops 😅

**Comment by stefanorigano** [comment]
Hello @Brieuc-brd , thanks so much for your review 👍 

> I also found a very specific bug: If you scroll enough to trigger the sticky position of the title, then change its background: this will remove the top calculation preventing the title from being hidden by the fixed header 🤔 :

I succeeded avoiding the issue by using a **non** `div.row > div.col` structure; apparently using `<div>` tags was triggering some JS invalidating the `top` computation when changing background-color indeed. I opted for:

```xml
<article class="row">
    <hgroup class="col-lg-4"/>
    <span class="d-block col-lg-7 offset-lg-1"/>
</article>
```
... that, in retrospective, makes a lot of sense sense by a semantic point of view (except for the `<span>` maybe, but we have to trade something I guess 🥲 ). Beside that:

- Added an option to create a new entry from the sidebar
- Hide the default option to force the snippet' height (not needed in this case).

Thanks!

File: addons/website/views/snippets/s_faq_horizontal.xml
Comment (Brieuc-brd) [inline addons/website/views/snippets/s_faq_horizontal.xml:22]:
```suggestion
                        <span data-name="Topic Content" class="col-lg-7 offset-lg-1 d-block">
```

File: addons/website/views/snippets/s_faq_horizontal.xml
Comment (Brieuc-brd) [inline addons/website/views/snippets/s_faq_horizontal.xml:39]:
```suggestion
                        <span data-name="Topic Content" class="col-lg-7 offset-lg-1 d-block">
```

File: addons/website/views/snippets/s_faq_horizontal.xml
Comment (Brieuc-brd) [inline addons/website/views/snippets/s_faq_horizontal.xml:60]:
```suggestion
                        <span data-name="Topic Content" class="col-lg-7 offset-lg-1 d-block">
```

**Review by Brieuc-brd — CHANGES_REQUESTED** [review]
Thanks for the update @stefanorigano 👍 

> I succeeded avoiding the issue by using a non div.row > div.col structure

Unfortunately, I think the issue persists 🙁 : 
https://www.awesomescreenshot.com/video/30778431?key=22196cbfddf5c5f67c070989d025215b
The strange thing is that when you save and go back in edit mode the issue seems to be gone .. until you save again.
Perhaps it has something to do with the `row` class used as a specific selector that removes the `top` property? 

> Added an option to create a new entry from the sidebar

This is a nice and necessary feature, thanks for adding it 👍
My only concern is that as this snippet is quite high, when you add an element while you're at the top of the snippet, you don't see the new element appear and so “nothing happens”. 
It would be nice to have an anchor that automatically scrolls to the new element, for example.
But I suppose that's the kind of improvement we can make later. I'll let you decide :)

About the latest changes you've made in the code, I have nothing to say except a few minor nitpicks I've left below 😬 

This snippet is not quite easy to handle, but you made a good job 👍 
Thanks and courage ! 💪

**Comment by stefanorigano** [comment]
Ok @Brieuc-brd , I think that the issue was caused by the fact that the JS was bind to each single entry rather than to the main snippet's body, we should be good now.
If there are still glitch we'll solve them during tech. 👍 

https://github.com/user-attachments/assets/738876b0-1d9f-41dd-a17c-251bea8b0e4a

> It would be nice to have an anchor that automatically scrolls to the new element, for example.

Yep, I wanted to do it as well. I found a bit of time today, done 👍 🙂

Moving this tech. Thank you !

File: addons/website/views/snippets/s_faq_horizontal.xml
Comment (bso-odoo) [inline addons/website/views/snippets/s_faq_horizontal.xml:46]: This is removed as soon as it is loaded - typically it is already removed inside the snippets dialog, before drop.
Wouldn't the `s_dialog_preview` class do the trick instead ?

File: addons/website/static/src/img/snippets_previews/s_faq_horizontal_2_preview.jpg
Comment (bso-odoo) [inline addons/website/static/src/img/snippets_previews/s_faq_horizontal_2_preview.jpg:1]: This will not match the palette, but I guess there is no simple solution for this.

**Review by bso-odoo — COMMENTED** [review]
Here are some comments:

File: addons/website/views/snippets/s_faq_horizontal.xml
Comment (stefanorigano) [reply addons/website/views/snippets/s_faq_horizontal.xml:46]: Indeed, I've definitely overthinked it ^^ 
Thanks 👍

File: addons/website/views/snippets/s_faq_horizontal.xml
Comment (stefanorigano) [reply addons/website/views/snippets/s_faq_horizontal.xml:64]: It should not indeed  😕
I'll remove one of the two buttons, I don't see any other solution. Thanks 👍

File: addons/website/static/src/img/snippets_previews/s_faq_horizontal_2_preview.jpg
Comment (stefanorigano) [reply addons/website/static/src/img/snippets_previews/s_faq_horizontal_2_preview.jpg:1]: Not that I know indeed.. but it should be fine for a demo preview indeed.

**Comment by stefanorigano** [comment]
@bso-odoo , suggestions applied and branch rebased, thanks 👍

File: addons/website/views/snippets/s_faq_horizontal.xml
Comment (bso-odoo) [inline addons/website/views/snippets/s_faq_horizontal.xml:18]: When I assign the style using the editor's toolbar, I get `h5-fs` within a nested `<span>`.

Comment (stefanorigano) [reply addons/website/views/snippets/s_faq_horizontal.xml:18]: Yes but... using a nested `<span>` will invalidate user edition. Are we sure that's the expected behaviour ?

https://github.com/user-attachments/assets/b7477386-972f-4864-bc4e-64b56d29adf4

Comment (qsm-odoo) [reply addons/website/views/snippets/s_faq_horizontal.xml:18]: @bso-odoo No nested span, let's put the class on the hx tags: https://discord.com/channels/678381219515465750/818823260967010304/1278692494451675238

Comment (bso-odoo) [reply addons/website/views/snippets/s_faq_horizontal.xml:18]: Both lead to the same situation after being edited, so, it's fine by me the way you did.
The editor side will definitely have to be fixed.

**Comment by bso-odoo** [comment]
LGTM, queued after `s_big_number` #175920

**Comment by bso-odoo** [comment]
@robodoo delegate=bvr-odoo

**Comment by stefanorigano** [comment]
_note: rebased and conflicts solved_

**Comment by bvr-odoo** [comment]
@robodoo r+

**Pull request closed without merging**
